### PR TITLE
Fixing OWASP so it doesn't inherit

### PR DESCRIPTION
**Pull request by michael-mclawhorn**
Moving to plugin management
Fixing cache property in screwdriver config

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: screwdriver.yaml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 cache:
-  event: ["~/.m2"]
+  job:
+    pull-commit: ["~/.m2"]
```
Comment (wcekan) [inline screwdriver.yaml:3]: Not necessary.  job cache is never set by pull request builds.
https://cd.screwdriver.cd/pipelines/6154/builds/683303/steps/sd-teardown-screwdriver-cache-bookend
> 00:00:00 2021/02/27 01:51:19 Skipping set job-scoped cache for Pull Request

However, using pipeline cache would let pull requests read the .m2 saved by the commit job.

File: screwdriver.yaml [hunk 2]
```diff
@@ -22,6 +24,7 @@ jobs:
       - debug: |
           echo "Master Commit : ${SD_EVENT_ID} ${SD_JOB_NAME} ${SD_PARENT_BUILD_ID}"
       - lua-install: apt-get update -qy && apt-get install -y lua5.2
+      - owasp-check: mvn -B org.owasp:dependency-check-maven:check
```
Comment (wcekan) [inline screwdriver.yaml:27]: Could we add `owasp-check` step to `pull-commit` as well?

File: pom.xml [hunk 3]
```diff
@@ -149,7 +149,7 @@
         <maven-project-info-reports-plugin-version>3.0.0</maven-project-info-reports-plugin-version>
         <maven-surefire-report-plugin-version>${maven-surefire-plugin-version}</maven-surefire-report-plugin-version>
 
-        <!-- empty property used as placeholder for args injected by jacoco coverage plugin -->
+        <version.owasp>6.1.0</version.owasp>
```
Comment (wcekan) [inline pom.xml:152]: It has been updated again
```suggestion
        <version.owasp>6.1.1</version.owasp>
```

**Comment by wcekan** [comment]
You may also want to add a teardown step to clean up the cache prior to storing it.  Much like the .travis build
https://github.com/yahoo/fili/blob/0bcc90be3ae8aa2182e1f747bbd51cb0f5da43d8/.travis.yml#L40

**Review by sgoplani — APPROVED** [review]

File: screwdriver.yaml (hunk 1 above)
Comment (wcekan) [reply screwdriver.yaml:3]: Let's also add the cache cleanup
```yaml
   - teardown-maven: >
        rm -rf /root/.m2/repository/com/yahoo/fili
```

**Pull request closed without merging**